lib/rspec/matchers/built_in/base_matcher.rb

```diff
@@ -96,6 +99,15 @@ def self.matcher_name
         end
 
         # @private
+        def matcher_name
+          if defined?(@matcher_name)
+            @matcher_name
+          else
+            self.class.matcher_name
```

Comment:
Why does this fall back to the class method (and is there a `matcher_name` class method even defined?).